addons/loyalty/models/loyalty_program.py

```diff
@@ -0,0 +1,318 @@
+    applies_on = fields.Selection([
+        ('current', 'Current order'),
+        ('future', 'Future orders'),
+        ('both', 'Current & Future orders')], default='current', required=True,
+         compute='_compute_from_program_type', readonly=False, store=True,
+    )
+    trigger = fields.Selection([
+        ('auto', 'Automatic'),
+        ('with_code', 'Use a code')],
+        compute='_compute_from_program_type', readonly=False, store=True,
+        help="""
+        Automatic: Customers will be eligible for a reward automatically in their cart.
+        Use a code: Customers will be eligible for a reward if they enter a code.
+        """
+    )
+    portal_visible = fields.Boolean(default=False,
+        help="""
+        Show in web portal, PoS customer ticket, eCommerce checkout, the number of points available and used by reward.
+        """)
+    portal_point_name = fields.Char(default='Points', translate=True,
+         compute='_compute_from_program_type', readonly=False, store=True)
+    is_nominative = fields.Boolean(compute='_compute_is_nominative')
+
+    _sql_constraints = [
+        ('check_max_usage', 'CHECK (limit_usage = False OR max_usage > 0)',
+            'Max usage must be strictly positive if a limit is used.'),
+    ]
+
+    @api.constrains('reward_ids')
+    def _constrains_reward_ids(self):
+        if any(not program.reward_ids for program in self):
+            raise ValidationError(_('A program must have at least one reward.'))
+
+    def _compute_total_order_count(self):
+        self.total_order_count = 0
+
+    @api.depends('company_id')
+    def _compute_currency_id(self):
+        for program in self:
+            program.currency_id = program.company_id.currency_id or program.currency_id
+
+    @api.depends('coupon_ids')
+    def _compute_coupon_count(self):
+        read_group_data = self.env['loyalty.card'].read_group([('program_id', 'in', self.ids)], ['program_id'], ['program_id'])
+        count_per_program = {r['program_id'][0]: r['program_id_count'] for r in read_group_data}
+        for program in self:
+            program.coupon_count = count_per_program.get(program.id, 0)
+
+    @api.depends('program_type', 'applies_on')
+    def _compute_is_nominative(self):
+        for program in self:
+            program.is_nominative = program.applies_on == 'both' or\
+                (program.program_type == 'ewallet' and program.applies_on == 'future')
+
+    @api.model
+    def _program_type_default_values(self):
+        # All values to change when program_type changes
+        # NOTE: any field used in `rule_ids`, `reward_ids` and `communication_plan_ids` MUST be present in the kanban view for it to work properly.
+        return {
+            'coupons': {
+                'applies_on': 'future',
+                'trigger': 'with_code',
+                'portal_visible': False,
+                'portal_point_name': 'Points',
```

Comment:
```suggestion
                'portal_point_name': _('Points'),
```
Same for the other terms.